### ISSUE TITLE
Handle ambiguous sticker names and values

### ISSUE THREAD

**Issue by h7x4** [MusicPlayerDaemon/MPD#2166]
There is an edgecase for stickers with names and values containing `=`, making the server output hard to parse.

```
$ nc localhost 6600
OK MPD 0.24.0
sticker set playlist TestPlaylist "test=1" "value=2"
OK
sticker list playlist TestPlaylist
sticker: test=1=value=2
OK
```

First solution that comes to mind would be to wrap the name and value in quotes, but then you could maybe put `"="` inside your string or something (I guess we could escape quotes as well?). Not sure what the best solution is here. Newlines maybe?

I feel like this is in the grey area of being a bug, considering it's up to the client developer to choose sensible tag names. But one client might interfere with another, so maybe it should be considered so.

**Comment by geneticdrift** [comment]
Adding for reference:
The issue of escaping `=` in sticker names was raised in the [PR](https://github.com/MusicPlayerDaemon/MPD/pull/1659) and was left out eventually.

This is the relevant part of a [longer comment](https://github.com/MusicPlayerDaemon/MPD/pull/1659#issuecomment-1562843824):
>>  "escape sticker name" - I'm not happy with this one. The escaping syntax for sticker names is not documented, and I'd rather restrict the allowed sticker names than adding some escaping rules. I mean, why would anybody really need = in a sticker name? Not restricting sticker names might have been an oversight in the initial design.

> Agree. Not a significant issue.

**Comment by jcorporation** [comment]
I have already some pull request for improved sticker handling open. After they are merged, I could look into this. I think restricting the sticker names is the best approach. Should we only disallow `=` or should we allow only `[a-zA-Z0-9]` and some special charakters like `.-_`?

**Comment by geneticdrift** [comment]
> restricting the sticker names is the best approach

Could be a  backward compatibility problem if sticker names that are already in the database become invalid.

So maybe just documenting the restrictions and not enforcing them?

From the docs:
> Client developers should create a standard for common sticker names, to ensure interoperability.

This never happened as far as I know, so clients may only be concerned with their own sticker names.

New restrictions can then be applied only to MPD officially standardized sticker names if and when that happens.

**Comment by jcorporation** [comment]
I had tried to create a standard: https://github.com/jcorporation/mpd-stickers

At the moment it is only a collection of used sticker namens.
